File: back-end/app/src/main/java/com/senescyt/app/model/ValorHora.java
package com.senescyt.app.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Timestamp;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "ValorHora")
public class ValorHora {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "valorId")
    private Long valorId;

    @Column(name = "valorHora")
    private double valorHora;

    @Column(name = "valorFecha")
    private Timestamp valorFecha;

    @Column(name = "valorEstado")
    private int valorEstado;
}

File: front-end/src/app/modelo/valorhora.ts
export class ValorHora {
  valorId: number;
  valorHora: number;
  valorFecha: Date;
  valorEstado: number;

  constructor(
    valorId?: number,
    valorHora?: number,
    valorFecha?: Date,
    valorEstado?: number
  ) {
    (this.valorId = valorId || 0),
      (this.valorHora = valorHora || 0),
      (this.valorFecha = valorFecha || new Date()),
      (this.valorEstado = valorEstado || 0);
  }
}
